Project/Gem/Source/Scripting/PayloadDespawnerComponent.cpp

```diff
@@ -0,0 +1,120 @@
+#include "Scripting/ScriptSpawnSytemBus.h"
+#include "std_msgs/msg/bool.hpp"
+#include <rclcpp/publisher.hpp>
+
+namespace ROS2::Demo
+{
+    void PayloadDespawnerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<PayloadDespawnerComponent>()->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<PayloadDespawnerComponent>("PayloadDespawnerComponent", "PayloadDespawnerComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2::Demo")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"));
+            }
+        }
+    }
+
+    void PayloadDespawnerComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("PhysicsColliderService"));
+    }
+
+    void PayloadDespawnerComponent::Activate()
+    {
+        m_onTriggerEnterHandler = AzPhysics::SimulatedBodyEvents::OnTriggerEnter::Handler(
+            [&]([[maybe_unused]] AzPhysics::SimulatedBodyHandle bodyHandle, [[maybe_unused]] const AzPhysics::TriggerEvent& event)
+            {
+                const auto entityId = event.m_otherBody->GetEntityId();
+                AZ::Entity* entity{};
+                AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+
+                if (IsObjectBox(entityId))
+                {
+                    AZ::Entity* entity{};
+                    AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+                    if (entity)
+                    {
+                        const auto& entityName = entity->GetName();
+                        ScriptSpawnSystemRequestBus::Broadcast(&ScriptSpawnSystemRequestBus::Events::DespawnBox, entityName);
+                    }
+                }
+                else if (IsObjectAMR(entityId))
+                {
+                    AZ::Entity* entity{};
+                    AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+                    auto frame = entity->FindComponent<ROS2FrameComponent>();
+                    AZ_Assert(frame, "AMR base_link does not have ROS2 Frame Component");
+                    AZStd::string amr_namespace = frame->GetNamespace();
+                    AZStd::string fullNamespace = amr_namespace + "/cargo_status";
+                    
+                    auto ros2Node = ROS2Interface::Get()->GetNode();
+                    auto deliberationPublisher =
+                        ros2Node->create_publisher<std_msgs::msg::Bool>(fullNamespace.data(), rclcpp::SensorDataQoS());
```

Comment:
Sensor QoS is not a good idea 
![image](https://github.com/RobotecAI/ROSCon2023Demo/assets/3209244/e17d0048-9da8-4ee4-9182-df20d00d9251)
It should be reliable.